compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -7087,4 +7091,32 @@ private void setClassQualifiers(NodeList<Token> qualifiers, BLangClassDefinition
             }
         }
     }
+
+    private List<BLangXMLStepExtend> createBLangXMLStepExtends(NodeList<Node> nodes, BLangExpression expr) {
+        List<BLangXMLStepExtend> extensions = new ArrayList<>();
+        BLangXMLStepExtend curExpr = null;
+        for (Node node : nodes) {
+            SyntaxKind kind = node.kind();
+            if (kind == SyntaxKind.XML_STEP_INDEXED_EXTEND) {
+                curExpr =
+                        new BLangXMLIndexedStepExtend(createExpression(((XMLStepIndexedExtendNode) node).expression()));
+            } else if (kind == SyntaxKind.XML_STEP_METHOD_CALL_EXTEND) {
+                XMLStepMethodCallExtendNode xmlStepMethodCallExtendNode = (XMLStepMethodCallExtendNode) node;
+                SimpleNameReferenceNode methodName = xmlStepMethodCallExtendNode.methodName();
+                BLangInvocation bLangInvocation = createBLangInvocation(methodName,
+                        xmlStepMethodCallExtendNode.parenthesizedArgList().arguments(), methodName.location(), false);
+                bLangInvocation.expr = curExpr == null ? expr : curExpr;
+                curExpr = new BLangXMLMethodCallStepExtend(bLangInvocation);
+            } else {
+                XMLNamePatternChainingNode xmlNamePatternChainingNode = (XMLNamePatternChainingNode) node;
+                List<BLangXMLElementFilter> filters = new ArrayList<>();
+                for (Node namePattern : xmlNamePatternChainingNode.xmlNamePattern()) {
+                    filters.add(createXMLElementFilter(namePattern));
+                }
+                curExpr = new BLangXMLFilterStepExtend(filters);
+            }
+            extensions.add(curExpr);
```

Comment:
let's switch on kind make the last branch explicit as well. Default can throw an error saying kind is invalid